modules/objdetect/src/qrcode.cpp

```diff
@@ -3753,18 +3822,24 @@ class ParallelDecodeProcess : public ParallelLoopBody
             else if (std::min(inarr.size().width, inarr.size().height) > 512)
             {
                 const int min_side = std::min(inarr.size().width, inarr.size().height);
-                double coeff_expansion = min_side / 512;
+                const float coeff_expansion = min_side / 512.f;
                 const int width  = cvRound(inarr.size().width  / coeff_expansion);
                 const int height = cvRound(inarr.size().height / coeff_expansion);
                 Size new_size(width, height);
                 Mat inarr2;
                 resize(inarr, inarr2, new_size, 0, 0, INTER_AREA);
-                for (size_t j = 0; j < 4; j++)
+                for (size_t j = 0ull; j < 4ull; j++)
                 {
-                    src_points[i][j] /= static_cast<float>(coeff_expansion);
+                    src_points[i][j] /= coeff_expansion;
                 }
                 qrdec[i].init(inarr2, src_points[i]);
                 ok = qrdec[i].straightDecodingProcess();
+                for (size_t j = 0ull; j < 4ull; j++) {
+                    src_points[i][j] = qrdec[i].getOriginalPoints()[j]*coeff_expansion;
+                }
+                for (size_t j = 0ull; j < qrdec[i].alignment_coords.size(); j++) {
+                    qrdec[i].alignment_coords[j] *= coeff_expansion;
```

Comment:
Are you sure that you need to break internal state if `qrdec[i]`? Is it used in the next steps? Most probably, you just need to scale the result in the next `if` block?